types/vote.go

```diff
@@ -202,14 +203,21 @@ func (vote *Vote) String() string {
 		panic("Unknown vote type")
 	}
 
-	return fmt.Sprintf("Vote{%v:%X %v/%02d/%v(%v) %X %X %X @ %s}",
+	var blockHashString string
+	if len(vote.BlockID.Hash) > 0 {
+		blockHashString = fmt.Sprintf("%X", tmbytes.Fingerprint(vote.BlockID.Hash))
+	} else {
+		blockHashString = nilVoteStr
+	}
+
+	return fmt.Sprintf("Vote{%v:%X %v/%02d/%v(%v) %s %X %X @ %s}",
 		vote.ValidatorIndex,
 		tmbytes.Fingerprint(vote.ValidatorAddress),
 		vote.Height,
 		vote.Round,
 		vote.Type,
 		typeString,
-		tmbytes.Fingerprint(vote.BlockID.Hash),
+		blockHashString,
 		tmbytes.Fingerprint(vote.Signature),
 		tmbytes.Fingerprint(vote.Extension),
```

Comment:
@thanethomson @williambanfield Looking at the string representation, it seems like we wanted votes to have quite a compact form. Adding the entire `vote.Extension` could change that. I'd opt to either remove it or to have a boolean that signals if it is present (I thought about including a hash but that's unnecessary computation - alternatively we could just include the first n bytes as a string).

Reply:
Adding on to @cmwaters' comment, consider including the extension's length

Reply:
As per suggestions, added length of the extension instead of its byte value.

0 would signify the absence, If vote extension is present, its byte length is printed

Reply:
Size (in bytes) seems like a good approach to me. A lot of applications of vote extensions use the field for non-human readable data like key shares etc, so including the first `n` bytes is likely of little utility.